Incremental output and isolated execution

GGmorello opened the pull request:
added incremental output to solve issue #2 
moved eval of the user's code to a separate worker
added a timeout to avoid infinite execution
refactored scripts from HTML to js files

GGmorello committed:
solved incremental output, added a web worker to execute user's code

GGmorello committed:
added timeout to stop execution

refactored code in external js file

jspenger commented:
I like the approach for now. Just some small outstanding issues.

Issue 1. The output size is not limited in size. It would make sense to limit the output to, say, 1024 rows, and truncate the oldest rows. See #2 .

Issue 2. The logging has a large overhead. For example, the following program (without logging) will execute within the time limit:
```javascript
var builder = PortalsJS.ApplicationBuilder("rangeFilter")
var _ = builder.workflows
  .source(builder.generators.fromRange(0, 1024*128, 8).stream)
  .filter(x => x % 1024 == 0)
  .logger()
  .sink()
  .freeze()
var rangeFilter = builder.build()
var system = PortalsJS.System()
system.launch(rangeFilter)
system.stepUntilComplete()
```
However, if we remove the filter then it will timeout around after 1024 events already. Perhaps we can make the logging more efficient. The first version I made was to simply append the new event, this may have a large overhead, perhaps. Perhaps you can try something, but if it takes too much time then it is good enough for now.

Issue 3.  Code location / distribution. I am not particularly fond of that the location of the new source files; nor that the examples are not together with the functions that set the editor to contain the example. Here are some initial ideas, but this perhaps requires more thought:
* Move the new JS files into `assets-playground/js/*`
* Create a new JS file `examples.js` containing the examples, perhaps move the `jsEditor.setValue` functions in there instead. 

This should be fine. I am also open for other suggestions.

Bonus:
Perhaps add a bonus example program that shows the incremental output. Something like this.
```javascript
var builder = PortalsJS.ApplicationBuilder("map")
function sleep(milliseconds) {
  const start = Date.now();
  while (Date.now() - start < milliseconds) {}
}
var _ = builder.workflows
  .source(builder.generators.fromRange(0, 8, 1).stream)
  .map(ctx => x => {
       sleep(200);
       return x;
  })
  .logger()
  .sink()
  .freeze()
var map = builder.build()
var system = PortalsJS.System()
system.launch(map)
system.stepUntilComplete()
```

jspenger reviewed (changes requested):
See above.

GGmorello committed:
refactored and optimized jsEditorRunner.js

refactored file structure moved examples to new file

added 100ms interval to display output, minor refactoring

added 1024 line limit, added scroll to end when lines don't fit the output textbox

GGmorello committed:
added 1024 line limit, added scroll to end when lines don't fit the output textbox

GGmorello committed:
added 100ms interval to display output, minor refactoring

GGmorello committed:
refactored file structure moved examples to new file

GGmorello committed:
refactored and optimized jsEditorRunner.js

GGmorello committed:
Merge remote-tracking branch 'origin/incremental-output' into incremental-output

jspenger commented:
LGTM
